gradio/blocks.py

```diff
@@ -1726,8 +1726,14 @@ async def postprocess_data(
 
             if block.stateful:
                 if not utils.is_update(predictions[i]):
-                    if block._id not in state or not utils.deep_equal(
-                        state[block._id], predictions[i]
+                    has_change_event = False
```

Comment:
@aliabid94 I implemented your suggestion to only compute the state diff of there is a change event defined for that state

Reply:
thanks, looks good!